LocalPackages/NetworkProtectionMac/Sources/NetworkProtectionUI/Views/TunnelControllerView/TunnelControllerViewModel.swift

```diff
@@ -140,29 +140,15 @@ public final class TunnelControllerViewModel: ObservableObject {
 
     private func subscribeToStatusChanges() {
         statusReporter.statusObserver.publisher
-            .subscribe(on: Self.statusDispatchQueue)
-            .sink { [weak self] status in
-
-            guard let self else {
-                return
-            }
-
-            Task { @MainActor in
-                self.connectionStatus = status
-                switch status {
-                case .connected, .connecting:
-                    self.isVPNEnabled = true
-                default:
-                    self.isVPNEnabled = false
-                }
-            }
-        }
+            .removeDuplicates()
+            .receive(on: DispatchQueue.main)
+            .assign(to: \.connectionStatus, onWeaklyHeld: self)
             .store(in: &cancellables)
     }
 
     private func subscribeToServerInfoChanges() {
         statusReporter.serverInfoObserver.publisher
-            .subscribe(on: Self.serverInfoDispatchQueue)
+            .receive(on: DispatchQueue.main)
```

Comment:
Moving all to the main dispatch queue because I was getting odd flickering in the status view.